src/plugins/auto/auto_schedule.cpp

```diff
@@ -560,29 +686,45 @@ bool AutoSchedule::ScheduleToWorkerInferRequest(IE::Task inferPipelineTask, Devi
     std::vector<DeviceInformation> devices;
     // AUTO work mode
     if (!preferred_device.empty()) {
-        // if the device needed by customer is not ready, need to wait for it
-        WaitActualNetworkReady();
-        // the preferred_device should be the selected device in AUTO work mode
-        if (preferred_device != _loadContext[ACTUALDEVICE].deviceInfo.deviceName) {
-            IE_THROW(NotFound) << "The preferred device should be the selected device";
+        if (_pCTPUTLoadContext) {
+            std::lock_guard<std::mutex> lock(_autoSContext->_fallbackMutex);
+            devices = _autoSContext->_devicePriorities;
+        } else {
+            // if the device needed by customer is not ready, need to wait for it
+            WaitActualNetworkReady();
+            // the preferred_device should be the selected device in AUTO work mode
+            if (preferred_device != _loadContext[ACTUALDEVICE].deviceInfo.deviceName) {
+                IE_THROW(NotFound) << "The preferred device should be the selected device";
+            }
+            devices.push_back(_loadContext[ACTUALDEVICE].deviceInfo);
```

Comment:
Why `_pCTPUTLoadContext` branch add lock, but not added here?

Reply:
you are right, lock is needed for fallback case

Reply:
@WeldonWangwang

Reply:
Because _autoSContext->_devicePriorities[i] will be modified when fallback in _pCTPUTLoadContext, so add lock to _devicePriorities for _pCTPUTLoadContext.